Mage.Sets/src/mage/cards/f/FurnacePunisher.java

```diff
@@ -0,0 +1,91 @@
+package mage.cards.f;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfUpkeepTriggeredAbility;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.condition.InvertCondition;
+import mage.abilities.condition.common.PermanentsOnTheBattlefieldCondition;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.keyword.MenaceAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.common.FilterLandPermanent;
+import mage.game.Game;
+import mage.players.Player;
+
+import java.util.UUID;
+
+public class FurnacePunisher extends CardImpl {
+
+    private static final FilterLandPermanent filter = new FilterLandPermanent();
+
+    static {
+        filter.add(SuperType.BASIC.getPredicate());
+    }
+
+    public FurnacePunisher(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{R}");
+        this.addSubType(SubType.PHYREXIAN);
+        this.addSubType(SubType.WARRIOR);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        //Menace
+        this.addAbility(new MenaceAbility(false));
+
+        //At the beginning of each player’s upkeep, Furnace Punisher deals 2 damage to that player unless they control
+        //two or more basic lands.
+        this.addAbility(new BeginningOfUpkeepTriggeredAbility(new FurnacePunisherEffect(),
+                TargetController.EACH_PLAYER,
+                false
+        ));
```

Comment:
Please apply `.setTriggerPhrase("At the beginning of each player's upkeep, ")` to `BeginningOfUpkeepTriggeredAbility` for desired rule text

Reply:
done